nuxeo-coldstorage/src/main/resources/OSGI-INF/coldstorage-events-contrib.xml

```diff
@@ -3,34 +3,44 @@
 
   <extension target="org.nuxeo.ecm.core.scheduler.SchedulerService" point="schedule">
     <schedule id="checkAvailabilityOfBlobsBeingRetrieved">
-      <!-- every hour at the 7th minute -->
-      <cronExpression>0 7 * ? * * *</cronExpression>
+      <!-- every 5 minute -->
+      <cronExpression>0 0/5 * ? * * *</cronExpression>
       <event>checkColdStorageContentAvailability</event>
     </schedule>
   </extension>
 
   <extension target="org.nuxeo.ecm.core.event.EventServiceComponent" point="listener">
-    <listener name="checkColdStorageContentAvailability"
+    <listener name="checkColdStorageContentAvailability" async="true"
               class="org.nuxeo.coldstorage.events.CheckColdStorageContentAvailabilityListener">
       <event>checkColdStorageContentAvailability</event>
     </listener>
     <listener name="checkUpdateMainContentInColdStorage"
               class="org.nuxeo.coldstorage.events.CheckUpdateMainContentInColdStorage">
       <event>beforeDocumentModification</event>
     </listener>
+    <listener name="checkColdStorageContentRestoreListener" async="true"
+              class="org.nuxeo.coldstorage.events.CheckColdStorageContentRestoreListener">
+      <event>coldStorageContentToRestore</event>
+    </listener>
   </extension>
 
   <extension
-    target="org.nuxeo.ecm.platform.ec.notification.service.NotificationService" point="notifications">
+          target="org.nuxeo.ecm.platform.ec.notification.service.NotificationService" point="notifications">
     <notification name="ColdStorageContentAvailable" channel="email" availableIn="*"
                   subject="Archive content available on '${docTitle}'" template="coldStorageContentAvailable"
                   label="label.document.download">
       <event name="coldStorageContentAvailable" />
     </notification>
+    <notification name="ColdStorageContentRestored" channel="email" availableIn="*"
+                  subject="Main content '${docTitle}' restored" template="coldStorageContentRestored"
+                  label="label.document.download">
+      <event name="coldStorageContentRestored" />
+    </notification>
   </extension>
 
   <extension target="org.nuxeo.ecm.platform.ec.notification.service.NotificationService" point="templates">
     <template name="coldStorageContentAvailable" src="templates/coldStorageContentAvailable.ftl" />
+    <template name="coldStorageContentRestored" src="templates/coldStorageContentRestored.ftl" />
   </extension>
 
 </component>
```

Comment:
Abdoul on this part could you please extract the part changing the listener type to `async` in a separate commit :) 

thx

Reply:
Hello @RSalem07 
we did the separation for the retrieve  ( it wasn't the case).
but the restore is a new feature, do we really need to split it into two commits?

Reply:
yes agree on that it was mainly for the check availability. thx